src/Properties/Currency.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace GingerPluginSdk\Properties;
+
+use GingerPluginSdk\Bases\BaseField;
+use GingerPluginSdk\Exceptions\OutOfEnumException;
+use GingerPluginSdk\Interfaces\ValidateFieldsInterface;
+
+final class Currency extends BaseField implements ValidateFieldsInterface
+{
+    /**
+     * @throws \GingerPluginSdk\Exceptions\OutOfEnumException
+     */
+    public function validate($value)
+    {
+        //@TODO: IMPLEMENT CACHING CURRENCY LIST AND USE THERE.
```

Comment:
I would leave currency validation to our API. If you want to provide a list of available currencies you could use the API for that.

Reply:
> I would leave currency validation to our API. If you want to provide a list of available currencies you could use the API for that.


Yes, i want to use API here. We already implemented in ginger-php library, from our side a method which is called getCurrencyList(). Here my comment is a note, as a wish, to implement caching the output of the request to API (caching will be implemented inside the library, and by this way, we can use multi-currency solutions more efficient inside all plugins which using the library). As we know, from Joachim, the currency list is not updating as many times as we requested. So prestroring response to currency list looks logical, is it the same for you?